RELEASE_GUIDE.md

```diff
@@ -44,48 +67,142 @@ considering what effects this may have:
 
 ## Making a release
 
-- Make a commit bumping versions. The following should be updated:
+A release is made by updating the versions within the repo via a PR. Once the PR is merged, a GitHub release is drafted, which triggers a CI build that should upload prebuilt binaries into that GitHub release page. (If it doesn't, you'll upload those manually).
```

Comment:
Can you avoid the passive voice for actions that the maintainer needs to take please? I think it is making it unclear which steps are performed automatically and which ones need to be done by the person making the release. Perhaps something like: "To make a release, a maintainer should create a PR updating versions in the repo, and once that is merged, they can create a GitHub release, which triggers a CI build ..."